src/nupic/algorithms/Connections.cpp

```diff
@@ -593,7 +596,7 @@ void Connections::load(std::istream &inStream) {
         inStream >> destroyedSegment;
       }
 
-      Segment segment = {(UInt32)-1};
+      Segment segment = {std::numeric_limits<Segment>::max()-1};
```

Comment:
-1 == std::numeric_limits::max()
You've got an extra -1.

Reply:
might have been the cause why the test stalled